packages/web/components/Dashboard/Post/Post.tsx

```diff
@@ -289,6 +294,80 @@ const Post = ({ post, currentUser, refetch }: IPostProps) => {
     },
   })
 
+  const [createPostClap, createPostClapResult] = useCreatePostClapMutation({
+    update(cache, { data }) {
+      if (data?.createPostClap) {
+        cache.modify({
+          id: `${post.__typename}:${post.id}`,
+          fields: {
+            claps: (existingClaps: PostClap[] = []) => {
+              return [...existingClaps, data.createPostClap]
+            },
+          },
+        })
+      }
+    },
+    onError: (error) => toast.error(error),
+  })
+
+  const createNewPostClap = () => {
+    createPostClap({
+      variables: { postId: post.id },
+      optimisticResponse: {
+        __typename: 'Mutation',
+        createPostClap: {
+          __typename: 'PostClap',
+          id: generateNegativeRandomNumber(),
+          author: {
+            __typename: 'User',
+            handle: currentUser!.handle,
+            name: currentUser?.name,
+            profileImage: currentUser?.profileImage,
+            id: currentUser!.id,
```

Comment:
The non-null assertion right next to an optional chain on the same object looks strange. If you're going to assert `currentUser` isn't null, it means accepting that the type system can not promise you this won't throw at runtime, so optional chaining is pointless (e.g. if `currentUser` is null or undefined, then line 323 will throw, the optional chain on 324 is irrelevant since the only case where it makes a difference is unreachable).

If you know for sure that current user can't be null here, then you can use the non-null assertion in all four cases

Reply:
Sounds good! i went ahead and added `|| !currentUser` to the disabled attribute on the button as well!